spynnaker/pyNN/models/neuron/population_neurons_machine_vertex.py

```diff
@@ -57,23 +57,28 @@ class PopulationNeuronsMachineVertex(
     """
 
     __slots__ = [
-        "__change_requires_neuron_parameters_reload",
         "__key",
         "__sdram_partition",
         "__ring_buffer_shifts",
         "__weight_scales",
-        "__slice_index"]
+        "__slice_index",
```

Comment:
consider pushing this code repeated three times to PopulationMachineNeurons

Reply:
Population machine neurons has no slots on purpose to avoid diamond inheritance, therefore every class that extends it needs something like this to provide the index required.